runtime/runtime/src/lib.rs

```diff
@@ -1097,32 +1114,45 @@ impl Runtime {
         };
         // Refund for the unused portion of the gas at the price at which this gas was purchased.
         let mut gas_balance_refund = safe_gas_to_balance(action_receipt.gas_price, gas_refund)?;
-        let mut gas_deficit_amount = 0;
+
+        let mut gas_deficit = 0;
+        let mut gas_surplus = 0;
+
+        let price_delta = current_gas_price.abs_diff(action_receipt.gas_price);
+        let value_delta = safe_gas_to_balance(price_delta, result.gas_burnt)?;
         if current_gas_price > action_receipt.gas_price {
-            // In a rare scenario, when the current gas price is higher than the purchased gas
-            // price, the difference is subtracted from the refund. If the refund doesn't have
-            // enough balance to cover the difference, then the remaining balance is considered
-            // the deficit and it's reported in the stats for the balance checker.
-            gas_deficit_amount = safe_gas_to_balance(
-                current_gas_price - action_receipt.gas_price,
-                result.gas_burnt,
-            )?;
-            if gas_balance_refund >= gas_deficit_amount {
-                gas_balance_refund -= gas_deficit_amount;
-                gas_deficit_amount = 0;
-            } else {
-                gas_deficit_amount -= gas_balance_refund;
-                gas_balance_refund = 0;
-            }
+            // price increased, burning resulted in a deficit
+            gas_deficit = value_delta;
```

Comment:
Somewhat preexisting, but variable naming is somewhat confusing here. Says `gas_deficit`/`gas_surplus` but the variables are of type `Balance` (i.e. NEAR, not gas.)

Reply:
Yeah, agreed, makes it harder to read than it has to be.

I'll not touch the variable names in the original code. But I think it's not better in the new method.